### DISCUSSION TITLE
How to automatically change Learning Rate if loss function is not improved?

### DISCUSSION THREAD

**Discussion by cemox35** [asteroid-team/asteroid#579] [Q&A] [answered]
Hello there,

This is my code:

```
loss = PITLossWrapper(singlesrc_neg_sisdr, pit_from="pw_pt")
optimizer = optim.Adam(model.parameters(), lr=1e-3)
system = System(model, optimizer, loss, train_loader, val_loader)
trainer = Trainer(max_epochs=EPOCHS,
                          resume_from_checkpoint=MODEL_PATH[i],
                          gpus=1,
                          precision=16,
                          callbacks=[EarlyStopping('val_loss')])
trainer.fit(system)
```
How can I change the LR if the output of the loss function in the validation set is not improved in three consecutive epochs?

**Comment by mpariente** [comment, marked as answer]
Have a look at the recipes, most of them have a learning rate scheduler set up.
